doc/Language/traps.pod6

```diff
@@ -921,46 +921,64 @@ of the assignment operators instead:
 
 =head1 Regexes
 
-=head2 C«<{$x}>» vs C«$($x)»: Implicit EVAL
-
-Sometimes you may need to match a generated string in a regex. This can be done
-using C<$(…)> or C«<{…}>» syntax:
-
-=for code
-my $x = ‘ailemac’;
-say ‘I ♥ camelia’ ~~ / $($x.flip) /; # OUTPUT: «｢camelia｣␤»
-say ‘I ♥ camelia’ ~~ / <{$x.flip}> /; # OUTPUT: «｢camelia｣␤»
-
-However, the latter only works I<sometimes>.
-
-Internally C«<{…}>» EVAL-s the given string inside an anonymous regex, while
-C<$(…)> lexically interpolates the given string. So C«<{…}>» immediately breaks
-with more complicated inputs. For example:
-
-=for code
-my $x = ‘ailemac#’;
-say ‘I ♥ #camelia’ ~~ / $($x.flip) /; # OUTPUT: «｢#camelia｣␤»
-#                    ⚠ ↓↓ WRONG ↓↓ ⚠
-say ‘I ♥ #camelia’ ~~ / <{$x.flip}> /;
-# OUTPUT:
-# ===SORRY!===
-# Regex not terminated.
-# at EVAL_0:1
-# ------> anon regex { #camelia}⏏<EOL>
-# Malformed regex
-# at EVAL_0:1
-# ------> anon regex { #camelia}⏏<EOL>
-#     expecting any of:
-#         infix stopper
-
-Therefore, try not to use C«<{}>» unless you really need EVAL.
-
-Note that even though EVAL is normally considered unsafe, in this case
-it is restricted to a set of safe operations (which is why it works
-without L<MONKEY-SEE-NO-EVAL|
-/language/pragmas#index-entry-MONKEY-SEE-NO-EVAL-MONKEY-SEE-NO-EVAL> pragma).
-In theory, careless use of C«<{}>» will only result in an exception being
-thrown, and should not introduce security issues.
+=head1 Interpolation constructs
+
+Perl 6 offers several constructs to generate regexes at runtime through
+interpolation (see their detailed description
+L<here|/language/regexes#Regex_interpolation>). When a thus generated regex
+contains only characters that match themselves, some of these constructs behave
```

Comment:
"character that match themselves" == "literal characters"? We already know them match themselves.

Reply:
The thing is that, somewhat surprisingly, the main regex page does not make a clear distinction between literal characters and metacharacters. The latter term does not even occur there at all, and so it is obviously not defined there either. Since I regard the use of undefined terminology as a deficiency (exactly the type that we're trying to fix (lexcially interpolates?)), I included a descriptive definition of metacharacters in this section.

The better option would probably be to first amend the main regex page to define the terms 'literal character' and 'metacharacter' properly, and then to rely on their definition here without an attempt to clarify them (again). If you agree, I'll make a concurrent proposal for such an amendment to the regex page.

Reply:
I just prepared a parallel PR to properly introduce the term 'metacharacter' in the regexes page. Once accepted, this PR can be cleaned up a bit.